pandas/tests/indexing/test_at.py

```diff
@@ -97,6 +99,18 @@ def test_at_setitem_categorical_missing(self):
 
         tm.assert_frame_equal(df, expected)
 
+    @pytest.mark.parametrize("row", (to_datetime("2019-01-01"), "2019-01-01"))
+    def test_at_datetime_index(self, row):
+        df = DataFrame(
+            data=[[1] * 2] * 2,
```

Comment:
Passing data=1 should work here. The dimensions are defined by columns and index in this case